app/routes/api/pokemon-types/$type.ts

```diff
@@ -0,0 +1,29 @@
+import type { ActionFunction, LoaderArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+import fs from 'fs';
+
+import pokemonList from '~/data/pokemon.json';
+
+export const loader = async ({ params }: LoaderArgs) => {
+	const filteredPokemon = pokemonList.pokemons.filter((pokemon) =>
+		pokemon.types.includes(params.type as string)
+	);
+	return json({
+		pokemons: filteredPokemon,
+	});
+};
+
+export const action: ActionFunction = async ({ request }) => {
+	if (request.method !== 'POST') return json({ error: 'Method not allowed' }, { status: 405 });
```

Comment:
in the action we can handle other methods as well. WDYT using a switch to handle other request.methods here? like for example PUT and DELETE